### PR TITLE
Ledger harvesting

### PR DESCRIPTION
**Pull request by AnthonyLaw**
### Fix
- Store vrf and remote key info before announce.
- unable to unlink a single key, caused by some [reactive properties](https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties) undefine, and assign a default value.

### CODE REVIEW

File: src/views/modals/ModalTransactionConfirmation/ModalTransactionConfirmationTs.ts [hunk 1]
```diff
@@ -742,6 +760,29 @@ export class ModalTransactionConfirmationTs extends Vue {
...
+            this.saveRemoteKeyInfo(
+                accountAddress,
+                Crypto.encrypt(this.command.newRemoteAccount.privateKey, this.command.password),
+                this.command.newRemoteAccount.publicKey,
+            );
+        }
+
+        // // pre-store selected harvesting node in local
+        this.saveHarvestingNode(accountAddress, this.command.formItems.nodeModel);
+
```
Comment (yilmazbahadir) [inline src/views/modals/ModalTransactionConfirmation/ModalTransactionConfirmationTs.ts:785]: This block looks like the block (L579-601), could we extract to a method and reuse that method?

File: src/views/forms/FormPersistentDelegationRequestTransaction/FormPersistentDelegationRequestTransactionTs.ts [hunk 2]
```diff
@@ -170,8 +170,8 @@ export class FormPersistentDelegationRequestTransactionTs extends FormTransactio
 
     private tempTransactionSigner: TransactionSigner;
     private tempAccount: Account;
-    public vrfPrivateKeyTemp: string;
-    public remotePrivateKeyTemp: string;
+    public vrfPrivateKeyTemp: string = '';
```
Comment (yilmazbahadir) [inline src/views/forms/FormPersistentDelegationRequestTransaction/FormPersistentDelegationRequestTransactionTs.ts:173]: Why are we initializing the temp keys?

Comment (yilmazbahadir) [reply src/views/forms/FormPersistentDelegationRequestTransaction/FormPersistentDelegationRequestTransactionTs.ts:173]: I guess that's the reason
> unable to unlink a single key, caused by some reactive properties undefine, and assign a default value.

File: src/views/modals/ModalTransactionConfirmation/ModalTransactionConfirmationTs.ts (hunk 1 above)
Comment (AnthonyLaw) [reply src/views/modals/ModalTransactionConfirmation/ModalTransactionConfirmationTs.ts:785]: yes, that is possible. Was thinking about where should we put this method.
HarvesrtingServices?

Comment (yilmazbahadir) [reply src/views/modals/ModalTransactionConfirmation/ModalTransactionConfirmationTs.ts:785]: I think a private method within the same file would be enough for now.

**Comment by cryptoBeliever** [comment]
It works now :+1: 
Checked:
- Ledger single account starting and stopping harvesting + manually removing links from "Key links" tab
- Ledger multi-sig account starting and stopping harvesting + manually removing links from "Key links" tab
- Regualr profile single harvesting
- Regular profile multisig harvesting

**Review by yilmazbahadir — APPROVED** [review]

**Pull request merged**